File: php/src/calisthenics-exercises/OnlyOneDotPerLine.php
<?php

namespace Calisthenics;

class Address
{
    public function __construct(
        public readonly string $street,
        public readonly string $city,
        public readonly string $zip
    ) {}

    public function getCity(): string
    {
        return $this->city;
    }
}

class Customer
{
    public function __construct(
        public readonly string $name,
        public readonly Address $address
    ) {}

    public function getAddress(): Address
    {
        return $this->address;
    }
}

class Order
{
    public function __construct(
        public readonly Customer $customer
    ) {}

    public function getCustomer(): Customer
    {
        return $this->customer;
    }

    public function getCustomerCity(): string
    {
        return $this->customer->getAddress()->getCity();
    }
}

File: python/src/calisthenics_exercises/only_one_dot_per_line.py
class Address:
    def __init__(self, street: str, city: str):
        self.street = street
        self.city = city


class Customer:
    def __init__(self, name: str, address: Address):
        self.name = name
        self.address = address


class Order:
    def __init__(self, customer: Customer, total: float):
        self.customer = customer
        self.total = total


def format_order_summary(order: Order) -> str:
    """Violación de la ley de Demeter: múltiples puntos por línea"""
    return f"Pedido de {order.customer.name} en {order.customer.address.city}"
